client/client_test.go

```diff
@@ -0,0 +1,256 @@
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strconv"
+	"syscall"
+	"testing"
+	"time"
+)
+
+// The IP adress of the test server
+const TestServerIP = "127.0.0.1"
+
+// The port that the server is listening on
+const TestServerPort = 8022
+
+// startTestServer starts a server listening at `TestServerIP` on
+// `TestServerPort`.  Need to later manually tear down the server.
+func startTestServer() (int, string, error) {
+	dir, err := ioutil.TempDir("", "TestServer")
+	if err != nil {
+		return 0, "", err
+	}
+
+	cmd := exec.Command("go",
+		"run",
+		fmt.Sprintf("%s/src/github.com/keybase/search-server/sserver/sserver/main.go", os.Getenv("GOPATH")),
+		fmt.Sprintf("--server_dir=%s", dir),
+		fmt.Sprintf("--port=%d", TestServerPort),
+		fmt.Sprintf("--ip_addr=%s", TestServerIP))
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	err = cmd.Start()
+	if err != nil {
+		os.RemoveAll(dir)
+		return 0, "", err
+	}
+
+	// Need to wait for the server to properly start
+	time.Sleep(time.Second * 10)
```

Comment:
Tests should run as fast as possible, so it's best to avoid arbitrary timeouts.  Using `rpc.Connection` will help avoid the need for this, I think.